cmd/metricsexporter/metrics_exporter.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"flag"
+	"net"
+	"net/http"
+	"os"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/promlog"
+	promflag "github.com/prometheus/common/promlog/flag"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+var (
+	exporterNamespace string
+	exporterAddress   string
+	kubeconfigPath    string
+	kubeAPIServerURL  string
+)
+
+type lvmCollector struct{}
+
+func (lc *lvmCollector) Describe(c chan<- *prometheus.Desc) {}
+func (lc *lvmCollector) Collect(c chan<- prometheus.Metric) {}
+
+func main() {
+	lvmExpFlags := flag.NewFlagSet("lvm-metrics-exporter", flag.ExitOnError)
+	lvmExpFlags.StringVar(&exporterNamespace, "namespace", "lvm-exporter",
+		"set the namespace of the lvm metric exporter")
+	lvmExpFlags.StringVar(&exporterAddress, "address", ":23532",
+		"address on which the metrics exporter should run")
+	lvmExpFlags.StringVar(&kubeconfigPath, "kubeconfig", "", "Path to kubeconfig file")
+	lvmExpFlags.StringVar(&kubeAPIServerURL, "apiserver", "", "API server URL")
+
+	zapOptions := &zap.Options{}
+	zapOptions.BindFlags(lvmExpFlags)
+
+	err := lvmExpFlags.Parse(os.Args[1:])
+	_ = err // we have ExitOnError handling set
+
+	zapOpts := zap.UseFlagOptions(zapOptions)
+	newLogger := zap.New(zapOpts).WithName("lvm-metric-exporter")
+	promlogConfig := &promlog.Config{}
+	promflag.AddFlags(kingpin.CommandLine, promlogConfig)
+	newLogger.Info("Commandline: ", "args", os.Args[1:])
+	newLogger.Info("Exporter set values: ", "namespace", exporterNamespace, "address", exporterAddress)
+
+	prometheus.MustRegister(&lvmCollector{})
+
+	// prometheus http handler
+	http.Handle("/metrics", promhttp.Handler())
+	http.HandleFunc("/", func(rw http.ResponseWriter, r *http.Request) {
+		_, err = rw.Write([]byte(`<html>
+             <head><title>LVM Metric Exporter</title></head>
+				<body>
+					<h1>LVM Metric Exporter</h1>
+					<p><a href='/metrics'>Metrics</a></p>
+				</body>
+             </html>`))
+		newLogger.Error(err, "error while writing into http.ResponseWriter")
+	})
+
+	// prometheus metrics
+	_ = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "testname",
+		Help: "this is a test gauge metrics",
+	})
```

Comment:
This is not being used. Either set some value or remove it?

Reply:
As we are using `promauto` package which injects this metric into the scrapped metrics' list. This is the way we check whether our lvm-operator's metric exporter is working or not (checking this value in the prometheus)

Reply:
It's added to the list, but does your metrics has any value set? The Collect() function seems empty.